compiler/rustc_middle/src/util/mod.rs

```diff
@@ -29,3 +34,13 @@ impl std::ops::Deref for Providers {
         &self.queries
     }
 }
+
+impl<'tcx> TyCtxt<'tcx> {
```

Comment:
Maybe put it to `compiler\rustc_middle\src\ty\context.rs`?
That's where `TyCtxt` is defined and many of its random inherent methods live.